rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -4524,7 +4528,12 @@ bool ppu_initialize(const ppu_module& info, bool check_only, u64 file_size)
 
 		const std::string dev_flash = vfs::get("/dev_flash/");
 
-		if (!info.path.starts_with(dev_flash) && !Emu.GetTitleID().empty() && Emu.GetCat() != "1P")
+		if (Emu.IsVsh())
+		{
+			// Add prefix for vsh
+			cache_path += "vsh/";
+		}
+		else if (!info.path.starts_with(dev_flash) && !Emu.GetTitleID().empty() && Emu.GetCat() != "1P")
```

Comment:
You should only add the directory prefix if it is inside dev_flash but  outside /dev_flash/sys/external

Reply:
When I think about it, dont check Emu.IsVsh(), just the previous conditions mentioned.

Reply:
I changed it in both functions